equinox/_filters.py

```diff
@@ -1,5 +1,15 @@
 from collections.abc import Callable
-from typing import Any, Final, overload, TypeVar
+from typing import (
+    Any,
+    Final,
+    Never,
+    overload,
+    runtime_checkable,
+    TypeAlias,
+    TypeGuard,
```

Comment:
Perhaps we should check for 3.13+ and use `TypeIs` in that case?

Reply:
I don't think that's what we want in this case. `TypeIs` doesn't enable type narrowing within the union.